kong/ids.go

```diff
@@ -0,0 +1,132 @@
+package kong
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/google/uuid"
+)
+
+// IDFillableEntitiesT is a type constraint for entities that can be filled with an ID.
+type IDFillableEntitiesT interface {
+	Service | Route | Consumer
+}
+
+// FillEntityID fills the ID of an entity. It is a no-op if the entity already has an ID.
+// ID is generated in a deterministic way using UUIDv5. The UUIDv5 namespace is different for each entity type.
+// The name used to generate the UUIDv5 is:
+// - for Service:    Service.Name
+// - for Route:      Route.Name
+// - for Consumer:   Consumer.Username
+func FillEntityID[T IDFillableEntitiesT](entity *T) error {
+	switch e := any(entity).(type) {
+	case *Service:
+		return fillIDForService(e)
+	case *Route:
+		return fillIDForRoute(e)
+	case *Consumer:
+		return fillIDForConsumer(e)
+	default:
+		return fmt.Errorf("unsupported entity: '%T'", entity)
+	}
+}
+
+func fillIDForService(s *Service) error {
```

Comment:
I see something else now. Would you consider making FillID an Entity method. For example :

```suggestion
func (s *Service) FillÎD() error {
```

Reply:
Yeah, that will also work. I was just trying to mimic the pattern that was used for `FillEntityDefaults` and `FillPluginsDefaults`.